### PR TITLE
Add Install tests

### PR DESCRIPTION
**Pull request by KyleFromKitware**
Fixes: #1166

### CODE REVIEW

File: testing/adios2/install/C/main.c [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/*
+ * Distributed under the OSI-approved Apache License, Version 2.0.  See
+ * accompanying file Copyright.txt for details.
+ */
+
+#include <adios2_c.h>
+
+int main(void)
+{
+  return 0;
```
Comment (chuckatkins) [inline testing/adios2/install/C/main.c]: Let's go ahead and add a symbol to validate linking.  It doesn't need to even be working code, so long as it compiles, links, and doesn't segfault.
```C
int main(void)
{
  adios2_finalize(NULL);
  return 0;
}
```
@williamfgc would this snippet execute without segfaulting?

File: testing/adios2/install/CXX11/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#------------------------------------------------------------------------------#
+# Distributed under the OSI-approved Apache License, Version 2.0.  See
+# accompanying file Copyright.txt for details.
+#------------------------------------------------------------------------------#
+
+cmake_minimum_required(VERSION 3.6)
+project(adios_c_test CXX)
```
Comment (chuckatkins) [inline testing/adios2/install/CXX11/CMakeLists.txt]: Should `project(adios_c_test CXX)` be `project(adios_cxx11_test CXX)`?

File: testing/adios2/install/CXX11/main.cxx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+/*
+ * Distributed under the OSI-approved Apache License, Version 2.0.  See
+ * accompanying file Copyright.txt for details.
+ */
+
+#include <adios2.h>
+
+int main(void)
+{
```
Comment (chuckatkins) [inline testing/adios2/install/CXX11/main.cxx]: Same as with C:
```cxx
int main(int, char **)
{
  adios2::ADIOS adios;
  return 0;
}
```

File: testing/adios2/install/CXX11/CMakeLists.txt (hunk 2 above)
Comment (KyleFromKitware) [reply testing/adios2/install/CXX11/CMakeLists.txt]: Indeed, good catch

**Comment by KyleFromKitware** [comment]
> Would it be easier to have the test use ExternalProject? That way you can easily add the manual makefile tests as well and do it in the same way as the cmake based test?

`ctest --build-and-test` has the ability to build a project with your own configure and build steps (or with no configure step.) `ExternalProject` would build the test project at build time, and I would rather build it at test time.

**Pull request merged**